### PR TITLE
Fix `update_fees()`, the calculation is wrong (refund)

### PR DESCRIPTION
**Pull request by rvola**
hello, this pull request correct the problem regarding a partial refund. The updated Stripe fee is not correct. This corrects him.

### CODE REVIEW

**Comment by roykho** [comment]
Can you please explain why this is needed? The API endpoint we're hitting is already from the balance history. When it returns, it "should" give us back the correct final balance no?

**Comment by rvola** [comment]
Not precisely, the fee only corresponds to the current transaction. If you make a partial refund, you update the post_meta with the commission of the partial refund, or it does not correspond to the total commission on the order

**Comment by roykho** [comment]
Ok thanks for explaining. I will check with Stripe to see if there is another endpoint that gets the true net total of fees. If not, then we'll proceed with this PR review.

**Comment by rvola** [comment]
I will add that it was when I worked on the request #527 , that I realized that the calculation is wrong.

File: includes/abstracts/abstract-wc-stripe-payment-gateway.php [hunk 1]
```diff
@@ -655,8 +655,16 @@ public function update_fees( $order, $balance_transaction_id ) {
 			if ( isset( $balance_transaction ) && isset( $balance_transaction->fee ) ) {
 				// Fees and Net needs to both come from Stripe to be accurate as the returned
 				// values are in the local currency of the Stripe account, not from WC.
-				$fee = ! empty( $balance_transaction->fee ) ? WC_Stripe_Helper::format_balance_fee( $balance_transaction, 'fee' ) : 0;
-				$net = ! empty( $balance_transaction->net ) ? WC_Stripe_Helper::format_balance_fee( $balance_transaction, 'net' ) : 0;
+				$fee_refund = ! empty( $balance_transaction->fee ) ? WC_Stripe_Helper::format_balance_fee( $balance_transaction, 'fee' ) : 0;
+				$net_refund = ! empty( $balance_transaction->net ) ? WC_Stripe_Helper::format_balance_fee( $balance_transaction, 'net' ) : 0;
+
+				/*current data fee*/
```
Comment (roykho) [inline includes/abstracts/abstract-wc-stripe-payment-gateway.php]: Coding standards: single line comments uses `//` and capital and also hard stop period at the end.

File: includes/abstracts/abstract-wc-stripe-payment-gateway.php [hunk 2]
```diff
@@ -655,8 +655,16 @@ public function update_fees( $order, $balance_transaction_id ) {
...
+				$fee_current = get_post_meta( $order_id, self::META_NAME_FEE, true );
+				$net_current = get_post_meta( $order_id, self::META_NAME_NET, true );
+
+				/*calculation*/
```
Comment (roykho) [inline includes/abstracts/abstract-wc-stripe-payment-gateway.php]: Coding standards: single line comments uses `//` and capital and also hard stop period at the end.

File: includes/abstracts/abstract-wc-stripe-payment-gateway.php [hunk 3]
```diff
@@ -655,8 +655,16 @@ public function update_fees( $order, $balance_transaction_id ) {
 			if ( isset( $balance_transaction ) && isset( $balance_transaction->fee ) ) {
 				// Fees and Net needs to both come from Stripe to be accurate as the returned
 				// values are in the local currency of the Stripe account, not from WC.
-				$fee = ! empty( $balance_transaction->fee ) ? WC_Stripe_Helper::format_balance_fee( $balance_transaction, 'fee' ) : 0;
-				$net = ! empty( $balance_transaction->net ) ? WC_Stripe_Helper::format_balance_fee( $balance_transaction, 'net' ) : 0;
+				$fee_refund = ! empty( $balance_transaction->fee ) ? WC_Stripe_Helper::format_balance_fee( $balance_transaction, 'fee' ) : 0;
+				$net_refund = ! empty( $balance_transaction->net ) ? WC_Stripe_Helper::format_balance_fee( $balance_transaction, 'net' ) : 0;
+
+				/*current data fee*/
+				$fee_current = get_post_meta( $order_id, self::META_NAME_FEE, true );
```
Comment (roykho) [inline includes/abstracts/abstract-wc-stripe-payment-gateway.php]: Need to use CRUD with BW compat.

File: includes/abstracts/abstract-wc-stripe-payment-gateway.php [hunk 4]
```diff
@@ -658,11 +658,11 @@ public function update_fees( $order, $balance_transaction_id ) {
 				$fee_refund = ! empty( $balance_transaction->fee ) ? WC_Stripe_Helper::format_balance_fee( $balance_transaction, 'fee' ) : 0;
 				$net_refund = ! empty( $balance_transaction->net ) ? WC_Stripe_Helper::format_balance_fee( $balance_transaction, 'net' ) : 0;
 
-				/*current data fee*/
-				$fee_current = get_post_meta( $order_id, self::META_NAME_FEE, true );
-				$net_current = get_post_meta( $order_id, self::META_NAME_NET, true );
+				// Current data fee & net
```
Comment (roykho) [inline includes/abstracts/abstract-wc-stripe-payment-gateway.php]: Need period at the end as commented before.

File: includes/abstracts/abstract-wc-stripe-payment-gateway.php [hunk 5]
```diff
@@ -658,11 +658,11 @@ public function update_fees( $order, $balance_transaction_id ) {
...
+				$fee_current = WC_Stripe_Helper::is_pre_30() ? get_post_meta( $order_id, self::META_NAME_FEE, true ) : $order->get_meta( self::META_NAME_FEE, true );
+				$net_current = WC_Stripe_Helper::is_pre_30() ? get_post_meta( $order_id, self::META_NAME_NET, true ) : $order->get_meta( self::META_NAME_NET, true );
 
-				/*calculation*/
+				// Calculation
```
Comment (roykho) [inline includes/abstracts/abstract-wc-stripe-payment-gateway.php]: Need period at the end as commented before.

File: includes/abstracts/abstract-wc-stripe-payment-gateway.php (hunk 4 above)
Comment (rvola) [reply includes/abstracts/abstract-wc-stripe-payment-gateway.php]: in order to understand, you just ask me to add a point (.) at the end of the comment?

Comment (roykho) [reply includes/abstracts/abstract-wc-stripe-payment-gateway.php]: Yes that is the coding standard.  See here

**Pull request merged**